src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -77,11 +77,10 @@ public void AddDependencyLocation(string fullPath)
                     _knownAssemblyPathsBySimpleName[simpleName] = paths.Add(fullPath);
                 }
 
-                // Ensure that there is no cached Assembly information about this location. Long 
-                // lived processes like VS and VBCSCompiler will see the same location added as
-                // a dependency many times. Each time have to assume there is new content on disk
-                // that needs to be considered.
-                _analyzerAssemblyInfoMap[fullPath] = null;
+                // This type assumses the file system is static for the duration of the
+                // it's instance. Repeated calls to this method, even if the underlying 
+                // file system contents, should reuse the results of the first call.
+                _ = _analyzerAssemblyInfoMap.TryAdd(fullPath, null);
```

Comment:
>_ = _analyzerAssemblyInfoMap.TryAdd(fullPath, null); [](http://example.com/codeflow?start=16&length=52)

Would it make sense to assert that the value in the dictionary after this is null?

Reply:
I think it's expected that the value will be non-null in many longer-lived loader scenarios.

It looks like the states of the dictionary correspond to roughly the following conditions:
- Dictionary does not contain key 'fullPath': A dependency has not yet been registered at 'fullPath'.
- Value is 'null' for key 'fullPath': The dependency has been registered, but we haven't loaded it yet.
- Value is non-null for key 'fullPath': The dependency has been registered, and loaded, and will be reused for the lifetime of the containing loader instance.

Reply:
A concrete example where it is non-null is a VS solution which has two projects that share analyzer references. Each of them, via the WorkSpaces layer, is going to call `AddDependency` for the same location on disk. The second one should be re-using the earlier one rather than overwriting it as my earlier change did.